maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/main/java/org/apache/maven/scm/provider/git/gitexe/command/GitCommandLineUtils.java

```diff
@@ -196,4 +209,27 @@ public static int execute( Commandline cl, CommandLineUtils.StringStreamConsumer
         return exitCode;
     }
 
+    static Map<String, String> prepareEnvVariablesForRepository( GitScmProviderRepository repository,
+                                                                 Map<String, String> environmentVariables )
+    {
+        Map<String, String> effectiveEnvironmentVariables = new HashMap<>();
+        if ( environmentVariables != null )
+        {
+            effectiveEnvironmentVariables.putAll( environmentVariables );
+        }
+        if ( StringUtils.isNotBlank( repository.getPrivateKey() ) )
+        {
+            if ( effectiveEnvironmentVariables.putIfAbsent( VARIABLE_GIT_SSH_COMMAND, "ssh -IdentitiesOnly=yes -i "
+                            + FilenameUtils.separatorsToUnix( repository.getPrivateKey() ) ) != null )
+            {
+                LOGGER.warn( "Ignore GitScmProviderRepository.privateKey as environment variable {} is already set",
+                             VARIABLE_GIT_SSH_COMMAND );
+            }
+        }
+        if ( StringUtils.isNotBlank( repository.getPassphrase() ) )
+        {
+            LOGGER.warn( "GitScmProviderRepository.passphrase currently not supported by provider 'git'" );
```

Comment:
gitexe is the term, no?

Reply:
Not sure. For me "gitexe" is more an internal term. It isn't mentioned in https://maven.apache.org/scm/git.html. I would stick to "git" as that is the provider name used in the SCM URL.